lib/game.rb

```diff
@@ -78,13 +77,12 @@ def attempt_fire_on_player_ship
     valid_range = @player_board.cells.keys
     valid_range.shuffle!
     coord = valid_range.shift
-    until !@player_board.cells[valid_range.first].fired_upon?
-      coord = valid_range.shift
-      valid_range.shuffle!
+    until !@player_board.cells[coord].fired_upon?
```

Comment:
We may still have too much logic here, but the issue was we were not using the new coord to validate if we could shoot or not.